login-module/src/index.js

```diff
@@ -1,19 +1,18 @@
-import constants from './constants';
-import { WebAuth } from 'auth0-js';
-import Quiz from './quiz';
-import Enrollment from './enrollment';
-import Certificate from './certificate';
-import misc from './misc';
-
-window.GraphAcademyLogin = class GraphAcademyLogin {
+import Auth0 from 'auth0-js';
+import constants from './constants.js';
+import Quiz from './quiz.js';
+import Enrollment from './enrollment.js';
+import Certificate from './certificate.js';
+import misc from './misc.js';
+
+export default class GraphAcademyLogin {
 	constructor(options = {}) {
 		this.options = { ...constants.DEFAULT_OPTIONS, ...options };
 		const hasRequiredOptions = this.hasRequiredOptions(options);
 		if (!hasRequiredOptions) {
-			console.log(`required params missing - one of ${constants.REQUIRED_OPTIONS.join(', ')}`);
-			return;
+			throw new Error(`required option missing - one of ${constants.REQUIRED_OPTIONS.join(', ')}`);
```

Comment:
I think it's better to fail fast.
If a required option is missing we should not return undefined but throw an Error.